src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -31,6 +34,37 @@ const SHA224: i64 = 224;
 const SHA256: i64 = 256;
 const SHA384: i64 = 384;
 const SHA512: i64 = 512;
+//const AES_BLOCK_SIZE: i64 = 16;
+
+fn derive_key_mysql(input: &[u8], block_size: usize) -> Vec<u8> {
+    let mut output: Vec<u8> = vec![0; block_size];
```

Comment:
Seems that `: Vec<u8>` is not necessary?

Reply:
Because &[u8] has length information, but AasRef<u8> does not. I need to traverse input.